atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/ConditionAwareTransactionTask.java

```diff
@@ -15,6 +15,6 @@
  */
 package com.palantir.atlasdb.transaction.api;
 
-public interface ConditionAwareTransactionTask<T, C extends PreCommitCondition, E extends Exception> {
+public interface ConditionAwareTransactionTask<T, C extends PreCommitConditionWithWatches, E extends Exception> {
```

Comment:
is this backwards compatible? There's no point not breaking if it's not

Reply:
i feel like the type erasure will end up with C being a PrecommitConditioNWithWatches, which will definitely be a binary breaking change.